docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,113 @@
+
+## Introduction
+
+In the current Fybrik implementation the entry point of a data plane is published as a part of the FybrikApplication 
+status, it is completely non-protected, and as result everyone can connect to the data plane. 
+In “Read” flow cases it leads to governance/security violations, “Write” and “Delete” flows can create severe accidents, 
+such as unauthorized data changes or deletions. Given that we need to prevent unauthorized data plane access. 
+
+_Note:_ the same situation exists between chained modules.  
+
+## Challenge
+
+When we talk about accessing Fybrik data plane, we are talking about connectivity between a user workload, which is out of 
+Fybrik scope, and Fybrik module, which can be developed by a third party. Therefore, in this document (and in its 
+extensions) we provide our recommendations how to protect data plane access and possible solutions.Responsibility of the
+data plane isolation is shared between Fybrik core, Fybrik users, module developers and probably IT teams.
+
+- Fybrik users should provide all relevant information to Fybrik and correctly configure their workloads. For example, 
+to set labels on the workload and inform Fybrik about the labels.
+
+- Depends on the isolation method, Fybrik should propagate all relevant information to the deployed modules, and/or 
+correctly configure Kubernetes Network Policies.
+- Module developers should allow to protect the module instances. For example, to allow TLS support
+
+In some complicated cases, IT teams interfere can be required, e.g. to configure external access points.
+
+## Requirements
+
+- Only predefined users/workloads should be able to access the data plane.
+- We do not assume that client workloads will be collocated on the same Kubernetes cluster with the data module, 
+furthermore they can run out of Kubernetes at all. 
+  - However, we do check the collocation of a Fybrik module and a relevant workload in the same cluster as a specific use case. 
+- We should allow workload client connections from different IP addresses.
+- Isolation of modules in the data plane. I.e. Modules should not be able to interface "illegally" with modules outside
+of the defined data plane(s) and connect to only predefined data sets.
+
+
+## Network Policies
+
+Fybrik is a Kubernetes application, therefore, the simplest isolation method can be based on 
+[Kubernetes Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) (NP).
+NP allow to control traffic flow at the IP address or port level (OSI layer 3 or 4). If a user workload or a Fybrik 
+previous chained module collocated with the next module in the same Kubernetes cluster, NPs can help to restrict the Fybrik 
+module ingress access. However, when a workload or a previous chained module runs out of the cluster, the NP based 
+isolation is going to be more complicated. The collocated deployment can be based on Kubernetes labels or namespaces, but 
```

Comment:
co-located (spelling)
I do not believe that Fybrik currently supports namesapace only (without a label) to identify a workload.  Namespace wouldn't be sufficient since there could be multiple workloads in one namespace.

Reply:
Except checking label existence for deprecated Flow definition, Fybrik do nothing with the labels.  (It's just copied into Plotter)

Reply:
I replaced collocated to co-located, but I've seen both forms as acceptable.